### ISSUE TITLE
[Bug] Importing deck with subdeck generated using genanki-js crash the app. Deck imported successfully on desktop

### ISSUE THREAD

**Issue by krmanik** [ankidroid/Anki-Android#9518]
###### Reproduction Steps

1. Visit [genanki-js/demo](https://infinyte7.github.io/genanki-js/demo/index.html#) page and select `Run example` from menu
2. Select any example from `Select test example` menu
3. Change deck name to `Test::New` for creating subdeck
```js
var d = new Deck(1276438724672, "Test::New")
```
4. Click `Run`, it generate Anki deck .apkg. 
5. Import in AnkiDroid 

The issue created in [genanki-js/issues](https://github.com/infinyte7/genanki-js/issues/4)

###### Expected Result
It should import deck with deck name `Test` and subdeck `New`. The .apkg file imported successfully on Anki desktop.

###### Actual Result
The app crashed and deck is not imported successfully. 

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

AnkiDroid = v2.16alpha14
Android Version = 11
Manufacturer = samsung
Scheduler = std2
Crash Reports Enabled = false
DatabaseV2 Enabled = true

###### Android Studio Logcat 
[logcat.txt](https://github.com/ankidroid/Anki-Android/files/7170802/logcat.txt)

###### Research
*Enter an [x] character to confirm the points below:*

- [ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ x ] I have searched for similar existing issues here and on the user forum
- [ x ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison-1** [comment]
[deck.zip](https://github.com/ankidroid/Anki-Android/files/7171686/deck.zip)

**Comment by david-allison-1** [comment]
Repro case for testing: [deck-repro-2.zip](https://github.com/ankidroid/Anki-Android/files/7171763/deck-repro-2.zip)

I need to add unit tests around import/export. 

Is this critical to get fixed by the next alpha?

**Comment by david-allison-1** [comment]
@infinyte7 This fails on 2.15.2 with an error message

and on the current alpha with a crash

**Comment by david-allison-1** [comment]
Cause: There is no default deck

https://github.com/ankidroid/Anki-Android/blob/db4ba1b042ffbb9677ee826e68421cdbb5ea25a5/AnkiDroid/src/main/java/com/ichi2/libanki/DeckManager.kt#L75

I've got a few PRs for the Rust conversion in.

I'll block this on https://github.com/ankidroid/Anki-Android/pull/9514 so we don't get conflicts

**Comment by infinyte7** [comment]
It is critical or not, depends upon original poster. I am not using subdeck creation currently. Because the error only occur for subdeck so it can be fixed later also. I will put remarks in my project so user can develop for deck only. When this is fixed I will removed the remarks.

**Comment by infinyte7** [comment]
It is found that the deck with subdeck generated using genanki python module import successfully on AnkiDroid. But using my project it is not imported. But on desktop it imported successfully. I have checked the the file difference between two sqlite dump db file to find the error in sql transaction.

I have extracted collection.anki2 sqlite db to text file compared the both file.

This is collection.anki2 dump db to text using genanki python module
[py.txt](https://github.com/infinyte7/genanki-js/files/7174601/py.txt)

This is collection.anki2 dump db to text using genanki-js 
[js.txt](https://github.com/infinyte7/genanki-js/files/7174602/js.txt)

The [diffchecker.com](https://www.diffchecker.com/) can be used to find the errors.

![image](https://user-images.githubusercontent.com/12841290/133542406-f91369e1-886d-431f-9e27-db588b356a5a.png)
